File: DataStructures/ExamPreparation-01/Problem-2-PitFortress/Java-Skeleton/PitFortress/src/main/PitFortressCollection.java
package main;

import main.interfaces.IPitFortress;
import main.models.Mine;
import main.models.Minion;
import main.models.Player;

public class PitFortressCollection implements IPitFortress {

    @Override
    public int getPlayerCount() {
        return 0;
    }

    @Override
    public int getMinionCount() {
        return 0;
    }

    @Override
    public int getMineCount() {
        return 0;
    }

    @Override
    public void addPlayer(String name, int mineRadius) {

    }

    @Override
    public void addMinion(int xCoordinate) {

    }

    @Override
    public void setMine(String playerName, int xCoordinate, int delay, int damage) {

    }

    @Override
    public Iterable<Minion> reportMinions() {
        return null;
    }

    @Override
    public Iterable<Player> top3PlayersByScore() {
        return null;
    }

    @Override
    public Iterable<Player> min3PlayersByScore() {
        return null;
    }

    @Override
    public Iterable<Mine> getMines() {
        return null;
    }

    @Override
    public void playTurn() {

    }
}

File: DataStructures/ExamPreparation-01/Problem-2-PitFortress/C#-Skeleton/PitFortressSkeleton/PitFortressCollection.cs
using System;
using System.Collections.Generic;
using Classes;
using Interfaces;
using Wintellect.PowerCollections;

public class PitFortressCollection : IPitFortress
{
        

    public PitFortressCollection()
    { 
    }

    public int PlayersCount { get; private set; }

    public int MinionsCount { get; private set; }

    public int MinesCount { get; private set; }

    public void AddPlayer(string name, int mineRadius)
    {
        throw new NotImplementedException();
    }

    public void AddMinion(int xCoordinate)
    {
        throw new NotImplementedException();
    }

    public void SetMine(string playerName, int xCoordinate, int delay, int damage)
    {
        throw new NotImplementedException();
    }

    public IEnumerable<Minion> ReportMinions()
    {
        throw new NotImplementedException();
    }

    public IEnumerable<Player> Top3PlayersByScore()
    {
        throw new NotImplementedException();
    }

    public IEnumerable<Player> Min3PlayersByScore()
    {
        throw new NotImplementedException();
    }

    public IEnumerable<Mine> GetMines()
    {
        throw new NotImplementedException();
    }

    public void PlayTurn()
    {
        throw new NotImplementedException();
    }
}
